pkg/controller/master/backup/backup_metadata.go

```diff
@@ -250,35 +249,31 @@ func (h *MetadataHandler) checkBackupBackingImageExist(imageMetadata *VirtualMac
 		return false
 	}
 
-	backingImageLabelReq, err := labels.NewRequirement("backing-image", selection.Equals, []string{backingImageName})
+	backupBackingImages, err := h.longhornBackupBackingImageCache.List(
+		util.LonghornSystemNamespaceName, labels.NewSelector())
 	if err != nil {
-		logrus.WithError(err).WithFields(logrus.Fields{
-			"namespace": imageMetadata.Namespace,
-			"name":      imageMetadata.Name,
-			"imageURL":  imageMetadata.URL,
-		}).Warn("Skip creating vm image, because cannot create new label requirement")
-		return false
-	}
-
-	if backupBackingImages, err := h.longhornBackupBackingImageCache.List(
-		util.LonghornSystemNamespaceName,
-		labels.NewSelector().Add(*backingImageLabelReq)); err != nil || len(backupBackingImages) == 0 {
 		logrus.WithError(err).WithFields(logrus.Fields{
 			"namespace": imageMetadata.Namespace,
 			"name":      imageMetadata.Name,
 			"imageURL":  imageMetadata.URL,
 		}).Warn("Skip creating vm image, because the backing image is not found")
 		return false
-	} else if backupBackingImages[0].Status.State != "Completed" {
-		logrus.WithError(err).WithFields(logrus.Fields{
-			"namespace": imageMetadata.Namespace,
-			"name":      imageMetadata.Name,
-			"imageURL":  imageMetadata.URL,
-		}).Warn("Skip creating vm image, because the backing image is not ready")
-		return false
+	}
+	for _, backupBackingImage := range backupBackingImages {
+		if backupBackingImage.Status.BackingImage == backingImageName {
+			if backupBackingImages[0].Status.State == "Completed" {
+				return true
+			}
+			logrus.WithError(err).WithFields(logrus.Fields{
```

Comment:
lgtm, Is the level error too high here?

Reply:
Hi @bk201, do you mean that using debug instead of warning?

Reply:
sorry my bad, it's warn level.